### DISCUSSION TITLE
Contract read data isn't re-fetched on page reload with Next.Js (caching?)

### DISCUSSION THREAD

**Discussion by xavierdmello** [wevm/wagmi#1780] [Q&A]
Hi, I have a useContractRead that just fetches data from a view function on a contract. However, I noticed that if a user closes the page and opens it again after the data changes, the page data isn't re-fetched - it just shows the old data. This is confusing to me because I thought that the useContractRead hooks are run on each page load.

I noticed that the data is refetched once the state the hook uses changes. In this case, it's a parameter in the useContractRead for an address. When the address state changes, the hook refetches. However, if the address is the same on different page loads, it doesn't refetch. Is this intended behavior? I want the data to be updated on each page load since the data is constantly changing on-chain and the user should see relatively updated info.

This only happens with data from wagmi hooks. It doesn't happen if I just use a regular ethers.js contract call in a useEffect hook. However, I want to use wagmi because of request deduplication.

I suspect that next.js or wagmi is caching data somewhere. Is there any way to turn this behavior off?

**Comment by IgorFox09** [comment]
https://wagmi.sh/react/hooks/useContractRead#watch-optional

**Reply by xavierdmello** [reply]
Watch refreshes on every block for me, and cacheTime/staleTime doesn't change that. Optimally it'd be simpler if it just did not cache data & refected on page reload, but I'd be happy to use watch if I could get it working. Any thoughts?

**Comment by jxom** [comment]
The data should revalidate (refresh) in the background. If you could open up an issue with a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) via a CodeSandbox or a repo, that would be great!
